project_wip/wip_to_cgs.py

```diff
@@ -0,0 +1,237 @@
+    )
+    wip_account_id = fields.Many2one(
+        "account.account",
+        "WIP Account",
+        company_dependent=True,
+        help="Account used to cumulate Work In Progress for this project type.",
+    )
+    cgs_account_id = fields.Many2one(
+        "account.account",
+        "CGS Account",
+        company_dependent=True,
+        help="Account used to cumulate Costs of Goods Sold for this project type.",
+    )
+
+    @api.constrains("wip_account_id")
+    def _check_wip_account_allows_reconcile(self):
+        """Check that the wip account on project type allows reconciliation."""
+        project_types_with_wip_accounts = self.filtered(lambda t: t.wip_account_id)
+        for project_type in project_types_with_wip_accounts:
+            if not project_type.wip_account_id.reconcile:
+                raise ValidationError(
+                    _(
+                        "The selected WIP account ({}) must allow reconciliation."
+                    ).format(project_type.wip_account_id.display_name)
+                )
+
+
+def _reconcile_wip_move_lines(wip_line, wip_reversal_line):
+    """Reconcile a WIP journal item with its reversal.
+
+    :param wip_line: the initial wip account.move.line
+    :param wip_reversal_line: the wip reversal account.move.line
+    :raises: ValidationError if the lines could not be reconciled.
+    """
+    unreconciled_line = (wip_line | wip_reversal_line).auto_reconcile_lines()
+    if unreconciled_line:
+        raise ValidationError(
+            _(
+                "The WIP entry {wip_line} ({amount}) could not be reconciled when transfering "
+                "the amount into Costs of Goods Sold. "
+                "You should verify if the WIP entry is partially reconciled."
+            ).format(wip_line=wip_line.display_name, amount=wip_line.balance)
+        )
+
+
+class Project(models.Model):
+
+    _inherit = "project.project"
+
+    def _check_project_type_has_wip_journal(self):
+        """Check that the project type has a wip journal defined.
+
+        :raises: ValidationError if no journal defined.
+        """
+        if not self.project_type_id.cgs_journal_id:
+            raise ValidationError(
+                _("The project type {} has no WIP journal defined.").format(
+                    self.project_type_id.name
+                )
+            )
+
+    def _check_project_type_has_wip_account(self):
+        """Check that the project type has a wip account defined.
+
+        :raises: ValidationError if no account defined.
+        """
+        if not self.project_type_id.wip_account_id:
+            raise ValidationError(
+                _(
+                    "The project type {} has no WIP (Work In Progress) account defined."
+                ).format(self.project_type_id.name)
+            )
+
+    def _check_project_type_has_cgs_account(self):
+        """Check that the project type has a cgs account defined.
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6165869)